sdk/cognitiveservices/azure-cognitiveservices-formrecognizer/DESIGN_DOC.md

```diff
@@ -0,0 +1,544 @@
+    lines: List[ExtractedLine]
+```
+
+
+### List/Get/Delete Models
+```python
+class ModelInfo:
+    model_id: str
+    status: str
+    created_on: ~datetime.datetime
+    last_updated_on: ~datetime.datetime
+
+class ModelsSummary:
+    count: int
+    limit: int
+    last_updated_on: ~datetime.datetime
+```
+
+
+### Custom Training Samples
+
+#### Custom: Train and Analyze without labels
+```python
+from azure.ai.formrecognizer import CustomFormClient
+
+client = CustomFormClient(endpoint=endpoint, credential=credential)
+
+# Train
+blob_sas_url = "xxxxx"  # training documents uploaded to blob storage
+poller = client.begin_training(blob_sas_url)
+model = poller.result()
+
+# Custom model information
+print("Model ID: {}".format(model.model_id))
+print("Status: {}".format(model.status))
+print("Created on: {}".format(model.created_on))
+print("Last updated on: {}".format(model.last_updated_on))
+
+# Training result information
+train_result = model.train_result
+for document in train_result.documents:
+    print(document.name)
+    print(document.status)
+    print(document.page_count)
+    print(document.errors)
+
+print("Form type ID: {}".format(train_result.form_type_id))
+print("Extracted fields:")
+print(train_result.extracted_fields)
+
+# Analyze
+blob_sas_url = "xxxxx"  # form to analyze uploaded to blob storage
+poller = client.begin_extract_forms(blob_sas_url, model_id=model.model_id)
+result = poller.result()
+
+for page in result:
+    print("Page: {}".format(page.page_number))
+    print("Form type ID: {}".format(page.form_type_id))
+    for field in page.fields:
+        print("{}: {}".format(field.name.text, field.value.text))
+        print("Confidence: {}".format(field.confidence))
+```
+
+#### Train and Analyze with labels
+```python
+from azure.ai.formrecognizer import CustomFormClient
+
+client = CustomFormClient(endpoint=endpoint, credential=credential)
+
+# Train
+blob_sas_url = "xxxxx"  # training documents uploaded to blob storage
+poller = client.begin_labeled_training(blob_sas_url)
+model = poller.result()
+
+# Custom model information
+print("Model ID: {}".format(model.model_id))
+print("Status: {}".format(model.status))
+print("Created on: {}".format(model.created_on))
+print("Last updated on: {}".format(model.last_updated_on))
+
+# Training result information
+train_result = model.train_result
+for document in train_result.documents:
+    print(document.name)
+    print(document.status)
+    print(document.page_count)
+    print(document.errors)
+
+print("Average model accuracy: {}".format(train_result.average_model_accuracy))
+print("Fields extracted/accuracy")
+for field in train_result.fields:
+    print(field.field_name, field.accuracy)
+
+blob_sas_url = "xxxxx"  # form to analyze uploaded to blob storage
+poller = client.begin_extract_labeled_forms(blob_sas_url, model_id=model.model_id)
+result = poller.result()
+
+print("Page range: {}".format(result.page_range))
+for field, value in result.labels.items():
```

Comment:
I know this is just sample code, but conceptually it might make more sense to think of the dictionary key as the "label" and the value as the "field"

Reply:
Fixed